core/src/main/java/com/novoda/rxpresso/mock/RxMocks.java

```diff
@@ -0,0 +1,240 @@
+package com.novoda.rxpresso.mock;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import rx.Notification;
+import rx.Observable;
+import rx.Subscriber;
+import rx.Subscription;
+import rx.functions.Action0;
+import rx.functions.Action1;
+import rx.functions.Func2;
+import rx.subjects.ClearableBehaviorSubject;
+import rx.subjects.PublishSubject;
+import rx.subscriptions.BooleanSubscription;
+
+import static com.novoda.rxpresso.mock.Functions.infinite;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+public final class RxMocks {
+
+    private final Object repository;
+    private final Map<String, Observable> observableHashMap = new HashMap<>();
+    private final Map<Observable, Pair<ClearableBehaviorSubject<Notification>, PublishSubject<Notification>>> mapSubject = new HashMap<>();
+
+    public static RxMocks mock(Class clazz) {
+        return init(Mockito.mock(clazz));
+    }
+
+    public static RxMocks init(Object repository) {
+        RxMocks rxMocks = new RxMocks(repository);
+        rxMocks.setMockResponses();
+        return rxMocks;
+    }
+
+    private RxMocks(Object repository) {
+        this.repository = repository;
+    }
+
+    private void setMockResponses() {
+        for (Method method : repository.getClass().getMethods()) {
+            if (method.getReturnType().equals(Observable.class) && isMockable(method)) {
+                setupMockResponseFor(method);
+            }
+        }
+    }
+
+    private boolean isMockable(Method method) {
```

Comment:
:+1:

For methods in interfaces, it would return `true` for `Modifier.isFinal(method.getModifiers)` which is why this check is omitted.